crates/fuel-core/src/schema/tx.rs

```diff
@@ -862,4 +880,40 @@ impl<'a> ContextExt for Context<'a> {
             }
         }
     }
+
+    async fn estimate_predicates(
+        &self,
+        mut tx: FuelTx,
+        query: impl PredicateStorageRequirements + Send + Sync + 'static,
+    ) -> anyhow::Result<FuelTx> {
+        let mut has_predicates = false;
+
+        for input in tx.inputs()? {
+            if input.predicate().is_some() {
+                has_predicates = true;
+                break;
+            }
+        }
+
+        if !has_predicates {
+            return Ok(tx);
+        }
+
+        let params = self
+            .data_unchecked::<ChainInfoProvider>()
+            .current_consensus_params();
+
+        let memory_pool = self.data_unchecked::<SharedMemoryPool>();
+        let memory = memory_pool.get_memory().await;
+
+        let parameters = CheckPredicateParams::from(params.as_ref());
+        let tx = tokio_rayon::spawn_fifo(move || {
+            let result = tx.estimate_predicates(&parameters, memory, &query);
+            result.map(|_| tx)
```

Comment:
Something like this should be more readable

```suggestion
            let _ = tx.estimate_predicates(&parameters, memory, &query)?;
            Ok(tx)
```